datasets/flwr_datasets/resplitter/divide_resplitter.py

```diff
@@ -0,0 +1,286 @@
+# Copyright 2024 Flower Labs GmbH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""DivideResplitter class for Flower Datasets."""
+import collections
+import warnings
+from typing import Dict, List, Optional, Union, cast
+
+import datasets
+from datasets import DatasetDict
+
+
+class DivideResplitter:
+    """Dive existing split(s) of the dataset and assign them custom names.
+
+    Create new `DatasetDict` with new split names corresponding the percentages of data
+    and custom names.
+
+    Parameters
+    ----------
+    divide_config: Union[Dict[str, int], Dict[str, float], Dict[str, Dict[str,
+    int]], Dict[str, Dict[str, float]]]
+        If single level dictionary with keys - the new split names and the values of int
+        = number of samples, float - fraction of the split. The fractions do not have
+        to sum up to 1.0. The order of matter = the first key will get fraction_1
+        starting from the beginning of the dataset.
+        If two level dictionary (dictionary of dictionaries) then the first keys are
+        the split names that will be divided into different splits. It's an alternative
+        to specifying `divide_split` if you need to divide many splits.
+    divide_split: Optional[str]
+        In case of single level dictionary specification of `divide_config`, specifies
+        the split name that will be divided. Might be left None in case of a single-
+        split dataset (it will be automatically inferred). Ignored in case of
+        multi-split configuration.
+    drop_remaining_splits: bool
+        In case of single level dictionary specification of `divide_config`, specifies
+        if the splits that are not divided are dropped.
+
+    Raises
+    ------
+    ValuesError if the specified name of a new split is already present in the dataset
+    and the `drop_remaining_splits` is False.
+    """
```

Comment:
I think it would be good to have some minimal example here, as some other components have in their docstrings body (e.g. `MergeSplitter`). Because different args types to `divide_config` results three different (but related) behaviours this would complement the description of the `Parameters` above.

Reply:
Done